src/README.md

```diff
@@ -0,0 +1,36 @@
+# Guideline for creation of new components
+When working on the implementation of new components please consider the internal guidelines below.
+
+## Component API design
+Before working on the implementation of a new component you are expected to think of a well-thought-out API design. 
+Have another contributor review your design to make sure nothing gets overlooked. Such a design includes:
+
+* Overview of **supported props**
+* Detailed definition of **prop types**. Go for the highest level of specificity that makes sense: don't just use `object` when you can use `shape`.
+* Which props are **required**
+* **Default** prop values for those that are not required
+* An **example** of conceived implementation, e.g.:
+
+```
+import { Button, ButtonGroup } from '@textkernel/nice';
+<ButtonGroup>
+    <Button
+        context="primary"
+        htmlType="link"
+        href="#"
+        target="_blank"
+        disabled
+    >
+        Click me
+    </Button>
+    ...
+</ButtonGroup>
+```
+
+## Definition of done
+Any new component should meet the following requirements before it can be merged.
+
+* The component is thorougly **covered by tests** (Jest / Enzyme), not only with snapshots
+* The component **comes with a story** in Storybook that allows props to be changed with knobs
+* Utils (functions) should be accompanied with **clear code comments**
+* The complete **build is passing**
```

Comment:
maybe add:
* code has been reviewed by at least one other developer
* component has been manually tested on different browsers (which ones?)

Reply:
Browser support and manual testing: good ones. We should point to the TK browser support guideline. Reviews by others are already mentioned, but I added it to the DoD points for sake of clarity.

Reply:
clear and  concise